src/components/GlobalSearch/globalSearch.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  CogniteClient,
+  Asset,
+  FilesMetadata,
+  CogniteEvent,
+} from '@cognite/sdk';
+// eslint-disable-next-line no-restricted-imports
+import { TimeSeries } from '@cognite/sdk/dist/src/resources/classes/timeSeries';
+import _ from 'lodash';
+
+export type SearchResource =
+  | AssetWithDiscriminator
+  | EventWithDiscrimnator
+  | TimeseriesWithDiscrimnator
+  | FileWithDiscrimnator;
+
+type AssetWithDiscriminator = Asset & {
+  resourceType: 'asset';
+};
+
+type TimeseriesWithDiscrimnator = TimeSeries & {
+  resourceType: 'timeSeries';
+};
+
+type FileWithDiscrimnator = FilesMetadata & {
+  resourceType: 'file';
+};
+
+type EventWithDiscrimnator = CogniteEvent & {
+  resourceType: 'event';
+};
+
+// Unsafe since it mutates the items of the array, to prevent unnecessary copies & method rebinds
+function unsafeWithDiscriminator<T>(
+  l: T[],
+  discriminator: string
+): SearchResource[] {
+  l.forEach(x => {
+    (x as any).resourceType = discriminator;
```

Comment:
could you instead wrap everything in 
```
type Resource = {
 item: Asset;
 type: 'asset';
} | { item: Timeseries, .... } | { ... }
```
this approach with any is breaking clean code

Reply:
Yeah that might be better